CMakeLists.txt

```diff
@@ -0,0 +1,213 @@
+# Support for building and installing the NIO demo programs with CMake
+#
+# CMake 3.15.0-rc3 is required for Swift support.
+# Ninja is also required.
+#
+# To build with CMake on Darwin
+#     mkdir build-darwin
+#     cd build-darwin
+#     xcrun cmake -GNinja -DCMAKE_BUILD_TYPE=RelWithDebugInfo -DBUILD_SHARED_LIBS:ON ..
+#     ninja
+#     # Create an installable package
+#     cpack -G TGZ .
+#     # Run integration tests
+#     ctest -VV .
+# To build with CMake on Linux
+#     docker-compose -f docker/docker-compose.yaml -f docker/docker-compose.1804.50.yaml up --build cmake-integration-tests
+cmake_minimum_required(VERSION 3.14.20190603)
+
+project(SwiftNIODemo LANGUAGES C Swift VERSION 2.3.0)
+
+include(GNUInstallDirs)
+
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib/SwiftNIODemo)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_Swift_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/swiftmodules)
+list(APPEND CMAKE_BUILD_RPATH "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
+
+add_compile_options($<$<COMPILE_LANGUAGE:Swift>:-warnings-as-errors>)
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  add_compile_options("$<$<COMPILE_LANGUAGE:Swift>:-target;x86_64-apple-macosx10.14>")
```

Comment:
This really shouldn't be needed.  `-DCMAKE_SYSTEM_NAME=Darwin` and `-DCMAKE_SYSTEM_PROCESSOR=x86_64` should do the right thing.  To pass the target explicitly, use `-DCMAKE_Swift_COMPILER_TARGET=`.